src/ui/flutter_app/lib/mill/position.dart

```diff
@@ -467,159 +467,141 @@ class Position {
-        return true;
-      }
+        // if illegal
+        if (pieceOnBoardCount[sideToMove]! > rule.flyPieceCount ||
+            !rule.mayFly) {
+          int md;
 
-      // if illegal
-      if (pieceOnBoardCount[sideToMove]! > rule.flyPieceCount || !rule.mayFly) {
-        int md;
+          for (md = 0; md < moveDirectionNumber; md++) {
+            if (s == adjacentSquares[currentSquare][md]) break;
+          }
 
-        for (md = 0; md < moveDirectionNumber; md++) {
-          if (s == adjacentSquares[currentSquare][md]) break;
+          // not in moveTable
+          if (md == moveDirectionNumber) {
+            debugPrint(
+              "[position] putPiece: [$s] is not in [$currentSquare]'s move table.",
+            );
+            return false;
+          }
         }
 
-        // not in moveTable
-        if (md == moveDirectionNumber) {
-          debugPrint(
-            "[position] putPiece: [$s] is not in [$currentSquare]'s move table.",
-          );
-          return false;
-        }
-      }
+        record =
+            "(${fileOf(currentSquare)},${rankOf(currentSquare)})->(${fileOf(s)},${rankOf(s)})";
 
-      record =
-          "(${fileOf(currentSquare)},${rankOf(currentSquare)})->(${fileOf(s)},${rankOf(s)})";
+        st.rule50++;
 
-      st.rule50++;
+        board[s] = _grid[squareToIndex[s]!] = board[currentSquare];
+        updateKey(s);
+        revertKey(currentSquare);
 
-      board[s] = _grid[squareToIndex[s]!] = board[currentSquare];
-      updateKey(s);
-      revertKey(currentSquare);
+        board[currentSquare] =
+            _grid[squareToIndex[currentSquare]!] = Piece.noPiece;
 
-      board[currentSquare] =
-          _grid[squareToIndex[currentSquare]!] = Piece.noPiece;
+        currentSquare = s;
+        final int n = millsCount(currentSquare);
 
-      currentSquare = s;
-      final int n = millsCount(currentSquare);
+        // midgame
+        if (n == 0) {
+          action = Act.select;
+          changeSideToMove();
 
-      // midgame
-      if (n == 0) {
-        action = Act.select;
-        changeSideToMove();
+          if (isGameOver()) return true;
 
-        if (isGameOver()) {
-          return true;
-        } else {
           gameInstance.focusIndex = squareToIndex[s] ?? invalidIndex;
           await Audios.playTone(Sound.place);
+        } else {
+          pieceToRemoveCount = rule.mayRemoveMultiple ? n : 1;
+          updateKeyMisc();
+          action = Act.remove;
+          gameInstance.focusIndex = squareToIndex[s] ?? invalidIndex;
+          await Audios.playTone(Sound.mill);
         }
-      } else {
-        pieceToRemoveCount = rule.mayRemoveMultiple ? n : 1;
-        updateKeyMisc();
-        action = Act.remove;
-        gameInstance.focusIndex = squareToIndex[s] ?? invalidIndex;
-        await Audios.playTone(Sound.mill);
-      }
-    } else {
-      assert(false);
+        break;
+      default:
+        assert(false);
     }
-
     return true;
```

Comment:
I think deleting the blank line above return will affect the appearance.

Reply:
see above (separate branch)